plugins/modules/aws_api_gateway_domain.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: aws_api_gateway_domain
+short_description: Manage AWS API Gateway custom domains
+description:
+     - Manages API Gateway custom domains for API GW Rest APIs.
+     - AWS API Gateway custom domain setups use CloudFront behind the scenes.
+       So you will get a CloudFront distribution as a result, configured to be aliased with your domain.
+version_added: '3.1.0'
+requirements: [ boto3 ]
+author:
+    - 'Stefan Horning (@stefanhorning)'
+options:
+  domain_name:
+    description:
+      - Domain name you want to use for your API GW deployment.
+    required: true
+    type: str
+  certificate_arn:
+    description:
+      - AWS Certificate Manger (ACM) TLS certificate ARN.
+    type: str
+    required: true
+  security_policy:
+    description:
+      - Set allowed TLS versions through AWS defined policies. Currently only TLS_1_0 and TLS_1_2 are available.
+      - Can only be set with botocore 1.12.175 or newer. On older boto versions this flag will automatically be ignored.
+    default: TLS_1_2
+    choices: ['TLS_1_0', 'TLS_1_2']
+    type: str
+  endpoint_type:
+    description:
+      - API endpoint configuration for domain. Use EDGE for edge-optimized endpoint, or use REGIONAL or PRIVATE.
+    default: EDGE
+    choices: ['EDGE', 'REGIONAL', 'PRIVATE']
+    type: str
+  domain_mappings:
+    description:
+      - Map your domain base paths to your API GW REST APIs, that you previously created. Use provided ID of the API setup and the release stage.
+      - "domain_mappings should be a list of dictionaries containing three keys: base_path, rest_api_id and stage."
+      - "Example: I([{ base_path: v1, rest_api_id: abc123, stage: production }])"
+      - if you want base path to be just I(/) omit the param completely or set it to empty string.
+    required: true
+    type: list
+    elements: dict
+  state:
+    description:
+      - Create or delete custom domain setup.
+    default: present
+    choices: [ 'present', 'absent' ]
+    type: str
+extends_documentation_fragment:
+    - amazon.aws.aws
+    - amazon.aws.ec2
+notes:
+   - Does not create a DNS entry on Route53, for that use the route53 module.
+   - Only supports TLS certificates from AWS ACM that can just be referenced by the ARN, while the AWS API still offers (deprecated)
+     options to add own Certificates.
+'''
+
+EXAMPLES = '''
+- name: Setup endpoint for a custom domain for your API Gateway HTTP API
+  aws_api_gateway_domain:
+    domain_name: myapi.foobar.com
+    certificate_arn: 'arn:aws:acm:us-east-1:1231123123:certificate/8bd89412-abc123-xxxxx'
+    security_policy: TLS_1_0
```

Comment:
```suggestion
    security_policy: TLS_1_2
```

example section should use state of the art (_in the aws world_) encryption. otherwise people copy/paste from the example section and uses and outdated TLS accidentally.